src/components/survey/other-structure.tsx

```diff
@@ -0,0 +1,311 @@
+import { Stack, Button, Typography, TextField, Accordion, AccordionSummary, AccordionDetails, Box, List } from "@mui/material";
+import Api from "api";
+import { useAppSelector } from "app/hooks";
+import { ErrorContext } from "components/error-handler/error-handler";
+import { selectKeycloak } from "features/auth-slice";
+import { Building, OtherStructure } from "generated/client";
+import strings from "localization/strings";
+import * as React from "react";
+import GenericDialog from "components/generic/generic-dialog";
+import WithDebounce from "components/generic/with-debounce";
+import produce from "immer";
+import { Add, ExpandMore } from "@mui/icons-material";
+
+/**
+ * Component properties
+ */
+interface Props {
+  surveyId: string;
+}
+
+/**
+ * Component for other structures
+ *
+ * @param props component properties
+ */
+const OtherStructures: React.FC<Props> = ({ surveyId }) => {
+  const keycloak = useAppSelector(selectKeycloak);
+  const errorContext = React.useContext(ErrorContext);
+  const [ building, setBuilding ] = React.useState<Building>();
+  const [ creatingOtherStructure, setCreatingOtherStructure ] = React.useState(false);
+  const [ deletingOtherStructureIndex, setDeletingOtherStructureIndex ] = React.useState<number>();
+  const [ newOtherStructure, setNewOtherStructure ] = React.useState({} as OtherStructure);
```

Comment:
Casting object like this might cause unwanted behavior since name and description are mandatory fields.
```suggestion
  const [ newOtherStructure, setNewOtherStructure ] = React.useState<OtherStructure>({ name: "", description: "" });
```